### PR TITLE
added benchmarks to phone_book example

### PR DESCRIPTION
**Pull request by bdemann**

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -84,6 +84,7 @@ jobs:
                     "examples/motoko_examples/hello-world",
                     "examples/motoko_examples/http_counter",
                     "examples/motoko_examples/minimal-counter-dapp",
+                    "examples/motoko_examples/phone-book",
```
Comment (lastmjs) [inline .github/workflows/test.yml:87]: This should be below `"examples/motoko_examples/persistent-storage"`

File: examples/motoko_examples/phone-book/canisters/azle/index.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
+import { ic, nat64, Opt, Query, Update } from 'azle';
+
+type PerfResult = {
```
Comment (lastmjs) [inline examples/motoko_examples/phone-book/canisters/azle/index.ts:3]: Comments around the performance code, talk to @dansteren

File: examples/motoko_examples/phone-book/canisters/azle/index.ts [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+}
+
+function record_performance(start: nat64, end: nat64): void {
+    perf_result = {
+        wasm_body_only: end - start,
+        wasm_including_prelude: ic.performance_counter(0)
+    };
+}
+
+export type Entry = {
```
Comment (lastmjs) [inline examples/motoko_examples/phone-book/canisters/azle/index.ts:21]: Is there a reason to export this type?

File: examples/motoko_examples/phone-book/canisters/declarations/index.js [hunk 4]
```diff
@@ -34,7 +34,7 @@ export const createActor = (canisterId, options) => {
 };
 
 /**
- * A ready-to-use agent for the phone_book canister
- * @type {import("@dfinity/agent").ActorSubclass<import("./phone_book.did.js")._SERVICE>}
+ * A ready-to-use agent for the azle canister
+ * @type {import("@dfinity/agent").ActorSubclass<import("./azle.did.js")._SERVICE>}
  */
-export const phone_book = createActor(canisterId);
+export const azle = createActor(canisterId);
```
Comment (lastmjs) [inline examples/motoko_examples/phone-book/canisters/declarations/index.js:40]: Should be commented out

File: examples/motoko_examples/phone-book/canisters/motoko/main.mo [hunk 5]
```diff
@@ -0,0 +1,47 @@
+import Map "mo:base/HashMap";
+import Prim "mo:⛔";
+import Text "mo:base/Text";
+
+actor {
+  type PerfResult = {
```
Comment (lastmjs) [inline examples/motoko_examples/phone-book/canisters/motoko/main.mo:6]: Comments around performance code

File: examples/motoko_examples/phone-book/canisters/rust/src/lib.rs [hunk 6]
```diff
@@ -0,0 +1,68 @@
+use std::cell::RefCell;
+use std::collections::HashMap;
+
+thread_local! {
```
Comment (lastmjs) [inline examples/motoko_examples/phone-book/canisters/rust/src/lib.rs:4]: Should be below the PerfResult struct

File: examples/motoko_examples/phone-book/canisters/rust/src/lib.rs [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+    static PERF_RESULT_REF_CELL: RefCell<Option<PerfResult>> = RefCell::default();
+}
+
+#[derive(candid::CandidType, Clone)]
+pub struct PerfResult {
```
Comment (lastmjs) [inline examples/motoko_examples/phone-book/canisters/rust/src/lib.rs:9]: Get the comments from @dansteren

File: examples/motoko_examples/phone-book/canisters/rust/src/lib.rs [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+    });
+
+    let perf_end = ic_cdk::api::call::performance_counter(0);
+
+    record_performance(perf_start, perf_end);
+}
+
+// TODO I don't think this default Entry is the right thing
+#[ic_cdk_macros::query]
+pub fn lookup(name: String) -> Entry {
```
Comment (lastmjs) [inline examples/motoko_examples/phone-book/canisters/rust/src/lib.rs:57]: This should return an `Option<Entry>`

Comment (lastmjs) [reply examples/motoko_examples/phone-book/canisters/rust/src/lib.rs:57]: I think this is the source of your problems the other day

**Review by lastmjs — CHANGES_REQUESTED** [review]

File: examples/motoko_examples/phone-book/canisters/azle/index.ts (hunk 3 above)
Comment (bdemann) [reply examples/motoko_examples/phone-book/canisters/azle/index.ts:21]: Yes the front end needs it to be exported

**Review by lastmjs — APPROVED** [review]

**Pull request merged**